rewrite/tests/python/all/format/tabs_and_indents_visitor_test.py

```diff
@@ -681,6 +681,77 @@ def my_function(a, b):
     return None
 
 
+def test_string_literal_assignment():
+    style = IntelliJ.tabs_and_indents().with_use_tab_character(False).with_tab_size(4)
+    rewrite_run(
+        # language=python
+        python(
+            '''
+             a = """
+             This is a string that
+             should not be modified.
+             """
+            '''
+        ),
+        spec=RecipeSpec().with_recipes(from_visitor(TabsAndIndentsVisitor(style)))
+    )
+
+
+def test_string_literal_assignment_in_function():
+    style = IntelliJ.tabs_and_indents().with_use_tab_character(False).with_tab_size(4)
+    rewrite_run(
+        # language=python
+        python(
+            '''
+            def my_function():
+             a = """
+             This is a string that
+             should align with the function body.
+             """
+             return None
```

Comment:
Actually, if you test this with PyCharm you will notice that it doesn't indent the contents of the string literal. This only happens when you remove the assignment, which identifies the string literal as a doc comment.

This is one area where Java text blocks behave differently. Their content is always stripped of all common leading indentation. As a result the text blocks can be indented without the string's runtime value changing. I think this is an extremely useful feature which I miss in Python. But that explains why the formatter must only indent the contents of doc comment string literals.

Reply:
I modified the test